workspaces/adventure-pack/goodies/typescript/Iterator.prototype.reduce/index.ts

```diff
@@ -0,0 +1,38 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    reduce<U>(
+      this: Iterator<T>,
+      callbackFn: (accumulator: U, value: T, index: number) => U,
+      initialValue?: U,
+    ): U;
+  }
+}
+
+iteratorPrototype.reduce ??= function <T extends U, U>(
+  this: Iterator<T>,
+  callbackFn: (accumulator: U, value: T, index: number) => T,
+  initialValue?: U,
+): U {
+  let index = 0;
+  let accumulator: U = initialValue!;
+  let accumulatorInitialized = arguments.length > 1;
```

Comment:
last nit: let's make boolean variables sound obviously boolean
I recommend using "is" or "does" as a great prefix for boolean variables that makes it very clear

```suggestion
  let isAccumulatorInitialized = arguments.length > 1;
```